iphone/Maps/Platform/opengl/iosOGLContextFactory.mm

```diff
@@ -38,6 +39,18 @@
 
 void iosOGLContextFactory::setPresentAvailable(bool available)
 {
+  lock_guard<mutex> lock(m_initializationMutex);
```

Comment:
I think it will be slightly more effective to do following:
```c++
{
  lock_guard lock;
  if (...)
  {
    m_isInitialized = true;
    ...
  }
}
if (m_drawContext ...)
  ...
```
i.e. to extract a critical section into an individual block. Up to you.